app/models/prison.rb

```diff
@@ -35,6 +35,10 @@ def last_bookable_date(today = Time.zone.today)
     today + booking_window
   end
 
+  def bookable_date?(requested_date = Time.zone.today)
+    available_slots.each(&:iso8601).include?(requested_date)
```

Comment:
I think a more expressive implementation might be to add a `to_date` method to `ConcreteSlot` and:

```ruby
available_slots.any? { |slot| slot.to_date == requested_date }
```

Or perhaps an `on?` method:

```ruby
available_slots.any? { |s| s.on?(requested_date) }
```